Tasks/AzureResourceGroupDeploymentV2/operations/ResourceGroup.ts

```diff
@@ -170,16 +170,71 @@ export class ResourceGroup {
             tl.error(error.message);
             if (error.details) {
                 tl.error(tl.loc("Details"));
+                let policyLink = null;
+
                 for (var i = 0; i < error.details.length; i++) {
-                    var errorMessage = util.format("%s: %s %s", error.details[i].code, error.details[i].message, error.details[i].details);
+                    var errorMessage = null;
+                    if (error.details[i].code === "RequestDisallowedByPolicy") {
+                        if (!policyLink) {
+                            policyLink = this.getPolicyHelpLink(error.details[i]);
+                        }
+
+                        errorMessage = this.getPolicyErrorMessage(error.details[i]);
+                    } else {
+                        errorMessage = util.format("%s: %s %s", error.details[i].code, error.details[i].message, error.details[i].details);
+                    }
+
                     tl.error(errorMessage);
                 }
+
+                if (policyLink) {
+                    tl.warning(tl.loc("ManageAzurePolicy") + policyLink);
+                }
             }
         } else {
             tl.error(error);
         }
     }
 
+    private getPolicyHelpLink(errorDetailMessage): string {
+        let managementGroupId = this.parseManagementGroupId(errorDetailMessage);
+        let portalUrl = this.taskParameters.endpointPortalUrl ? this.taskParameters.endpointPortalUrl : "https://portal.azure.com";
+        let policyLink = portalUrl + "#blade/Microsoft_Azure_Policy/PolicyMenuBlade/Assignments/scope/%2Fproviders%2FMicrosoft.Management%2FmanagementGroups%2F" + managementGroupId;
```

Comment:
This is generally not very good to put the UX links. The reason is that the Ux may change (unlike API). Also the task goes in on-prem as well where it is difficult to update the link

Reply:
Will close with the PM on this and will update you.